common/persistence/sql/sqlplugin/postgres/plugin.go

```diff
@@ -86,29 +83,46 @@ func (d *plugin) createDBConnection(cfg *config.SQL) (*sqlx.DB, error) {
 	}
 
 	dbName := cfg.DatabaseName
+	db, err := sqlx.Connect(PluginName, buildDSN(cfg, dbName, host, port))
+
+	if err != nil {
+		return nil, err
+	}
+
+	// Maps struct names in CamelCase to snake without need for db struct tags.
+	db.MapperFunc(strcase.ToSnake)
+	return db, nil
+}
+
+func buildDSN(cfg *config.SQL, dbName string, host string, port string) string {
 	//NOTE: postgres doesn't allow to connect with empty dbName, the admin dbName is "postgres"
 	if dbName == "" {
 		dbName = "postgres"
 	}
-	password := ""
+	params := url.Values{}
+	attrs := ""
+	
 	if cfg.Password != "" {
-		password = fmt.Sprintf(dataSourceNamePostgresPassword, cfg.Password)
+		params.Add("password", cfg.Password)
 	}
-	db, err := sqlx.Connect(PluginName, fmt.Sprintf(dataSourceNamePostgres, cfg.User, host, port, dbName, password))
 
-	if err != nil {
-		return nil, err
+	if cfg.TLS != nil && cfg.TLS.Enabled {
+		params.Add("ssl", "true")
+		params.Add("sslmode", "require")
+		params.Add("sslrootcert", cfg.TLS.CaFile)
+		params.Add("sslkey", cfg.TLS.KeyFile)
+		params.Add("sslcert", cfg.TLS.CertFile)
 	}
+	attrs = params.Encode()
 
-	// Maps struct names in CamelCase to snake without need for db struct tags.
-	db.MapperFunc(strcase.ToSnake)
-	return db, nil
+	dsn := fmt.Sprintf(dsnFmt, "postgresql", cfg.User, host, port, dbName)
+	if attrs != "" {
+		dsn += "?" + attrs
+	}
+	return dsn
 }
 
 // TODO: implement postgres specific support for TLS
```

Comment:
Should we remove the TODO?

Reply:
We could. The only reason I kept it in, because I thought it to be an interface implementation. 
Although I must say, I did a pass at temporal as well and the source files were slightly different, I might do another commit before going forward to synchronize the two directly, if that’s okay for you?

Reply:
Not quite follow exactly. Another commit for this? I thought this commit is working already. Or do you mean for other purpose.

Reply:
Sorry about the confusion, it does work, however I want to synchronize with https://github.com/temporalio/temporal/blob/master/common/persistence/sql/sqlplugin/postgres/plugin.go so I will do an additional commit tomorrow, before we go ahead with this one. I hope that makes sense.